src/common/components/EmbeddedSearch.jsx

```diff
@@ -0,0 +1,200 @@
+        page: 1,
+        size: 10,
+        sort: 'mostrecent',
+      },
+      numberOfResults: 0,
+      aggregations: Map(),
+      results: List(),
+      loadingResults: false,
+      loadingAggregations: false,
+    };
+  }
+
+  componentDidMount() {
+    this.searchForCurrentQueryState();
+  }
+
+  async onPageChange(page) {
+    this.updateQueryStateAndSearch({ page });
+  }
+
+  async onSortChange(sort) {
+    this.updateQueryStateAndSearch({ sort, page: 1 });
+  }
+
+  async onAggregationChange(aggregationKey, selections) {
+    this.updateQueryStateAndSearch({ [aggregationKey]: selections, page: 1 });
+  }
+
+  // eslint-disable-next-line react/sort-comp
+  async updateQueryStateAndSearch(partialQuery) {
+    const query = {
+      ...this.state.query,
+      ...partialQuery,
+    };
+    await this.setState({ query });
+    this.searchForCurrentQueryState();
+  }
+
+  searchForCurrentQueryState() {
+    this.fetchSearchResults();
+    this.fetchAggregations();
+  }
+
+  async fetchSearchResults() {
+    const { pidType } = this.props;
+    const queryString = stringify(this.getSearchQuery(), { indices: false });
+    const searchUrl = `/${pidType}?${queryString}`;
+    this.setState({ loadingResults: true });
+    try {
+      const { data } = await http.get(searchUrl);
+      this.setState({
+        results: fromJS(data.hits.hits),
+        numberOfResults: data.hits.total,
+        loadingResults: false,
+      });
+    } catch (error) {
+      this.setState({
+        hasError: true,
+        loadingResults: false,
+      });
+    }
+  }
+
+  async fetchAggregations() {
+    const { pidType, baseFacetsQuery } = this.props;
+    const query = {
+      ...baseFacetsQuery,
+      ...this.getSearchQuery(),
+    };
+    const queryString = stringify(query, { indices: false });
+    const searchUrl = `/${pidType}/facets?${queryString}`;
+    this.setState({ loadingAggregations: true });
+    try {
+      const { data } = await http.get(searchUrl);
+      this.setState({
+        aggregations: fromJS(data.aggregations),
+        loadingAggregations: false,
+      });
+    } catch (error) {
+      this.setState({
+        hasError: true,
+        loadingAggregations: false,
+      });
+    }
+  }
+
+  getSearchQuery() {
+    const { baseQuery } = this.props;
+    const { query } = this.state;
+    return mergeWithConcattingArrays(baseQuery, query);
+  }
+
+  renderErrorOrNull() {
+    const { hasError } = this.state;
+    if (hasError) {
+      return (
+        <Alert
+          message="Something went wrong, can not search publication of the author"
+          type="error"
+          showIcon
+          closable
+        />
+      );
+    }
+    return null;
+  }
+
+  render() {
+    const { renderResultItem } = this.props;
+    const {
+      query,
+      aggregations,
+      results,
+      numberOfResults,
+      loadingAggregations,
+      loadingResults,
+    } = this.state;
+    return (
+      this.renderErrorOrNull() || (
```

Comment:
shall we create a component to reuse it? We'll need for other places as well

Reply:
I don't think it's necessary, at the end it just an `Alert` and message is already specific for this use anyway.

Reply:
it's nice to use in places such as references, citations etc